lib/rspec/matchers/built_in/match.rb

```diff
@@ -5,6 +5,22 @@ module BuiltIn
       # Provides the implementation for `match`.
       # Not intended to be instantiated directly.
       class Match < BaseMatcher
+        # Used to specify the captures we match against
+        # Accepts variadic non-named captures
+        # @return [self]
+        def with_captures(*captures)
+          @expected_captures = captures
+          extend WithCaptures
+        end
```

Comment:
That example *might* sway me. However, it would require the code to be horribly mangled; the description would need to know not just that the match failed, but that it failed for this weird exceptional case. Which would require adding a bunch of additional state to the Match class in order to carry this knowledge from `match` to `description`. no?